src/jit/compiler.cpp

```diff
@@ -11085,3 +11085,231 @@ bool Compiler::killGCRefs(GenTree* tree)
     }
     return false;
 }
+
+//------------------------------------------------------------------------
+// getSiVarLoc: Returns a "siVarLoc" instance representing the place where the variable 
+// is given its description, "baseReg", and "offset" (if needed).
+//
+// Arguments:
+//    varDsc    - a "LclVarDsc *" to the variable it is desired to build the "siVarLoc".
+//    baseReg   - a "regNumber" use as a base for the offset.
+//    offset    - a signed amount of bytes distance from "baseReg" for the position of the variable.
+//
+// Return Value:
+//    A "siVarLoc" filled with the correct case struct fields for the variable, which could live
+//    in a register, an stack position, or a combination of both.
+//
+// Assumptions:
+//    No new memory is allocated on this method.
+//    No changes on "varDsc" are applied.
+//
+// Notes:
+//    The "baseReg" and "offset" cannot be obtained from "varDsc".
+//    It calls "fillRegisterVarLoc" or "fillStackVarLoc" depending on if the "varDsc" argument.
+//    "lvIsInReg", so their notes and assumptions are held.
+Compiler::siVarLoc Compiler::getSiVarLoc(LclVarDsc* varDsc, regNumber baseReg, signed offset) const
+{
+    siVarLoc  varLoc;
+    var_types type = genActualType(varDsc->TypeGet());
+
+    if (varDsc->lvIsInReg())
+    {
+        fillRegisterVarLoc(varDsc, varLoc, type, baseReg, offset);
+    }
+    else
+    {
+        fillStackVarLoc(varDsc, varLoc, type, baseReg, offset);
+    }
+
+    return varLoc;
+}
+
+//------------------------------------------------------------------------
+// fillStackVarLoc: Fill "varLoc" argument indicating the stack position of the variable
+// using "LclVarDsc" and "baseReg"/"offset".
+//
+// Arguments:
+//    varDsc    - a "LclVarDsc *" to the variable it is desired to build the "siVarLoc".
+//    varLoc    - a "siVarLoc &" to fill with the data of the "varDsc".
+//    type      - a "var_types" which indicate the type of the variable.
+//    baseReg   - a "regNumber" use as a base for the offset.
+//    offset    - a signed amount of bytes distance from "baseReg" for the position of the variable.
+//
+// Assumptions:
+//    No new memory is allocated on this method.
```

Comment:
I don't think I've ever seen anywhere in the JIT codebase such a mention being made in the function header. Also, it doesn't sound like an "assumption".

Reply:
I thought it would be usefull. Does it make more easy the search in case of a memory leak?
I can remove it.

Reply:
> Does it make more easy the search in case of a memory leak?

There's practically no way to leak memory in the JIT due to its use of an arena allocator.

Reply:
Good point! I will remove it. [Should we change the function header guide for ducumenting (part memory talks about memory to free)?](https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#94-function-header-comment)

Reply:
Are you talking about the "memory to be freed by caller" part? Yes, I think it should be removed from the guide. There are very few places in the JIT where memory should be freed and it's pretty much all debug code that use the host allocator.